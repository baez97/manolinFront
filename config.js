module.exports.BACKEND_IP = "http://192.168.43.205:5000";
// module.exports.BACKEND_IP = "http://192.168.1.38:5000";
// module.exports.BACKEND_IP = "http://192.168.1.33:5000";
// module.exports.BACKEND_IP = "http://192.168.1.205:5000";
// module.exports.BACKEND_IP = "http://192.168.1.35:5000";
// module.exports.BACKEND_IP = "https://manolin-backend.herokuapp.com";