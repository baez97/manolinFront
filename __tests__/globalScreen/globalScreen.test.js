import React from 'react';
import renderer from 'react-test-renderer';
import GlobalScreen from '../../screens/globalScreen';
import fetchMock from 'fetch-mock';
import mockedNurse from '../../mockedData/mockedNurse';
import {Â BACKEND_IP } from '../../config';

jest.unmock("react-native");
jest.unmock('react');
jest.unmock("react-test-renderer");
jest.unmock('expo');
jest.mock('../../components/dateUtils');
jest.useFakeTimers();

describe("GlobalScreen tests", () => {
    
    describe("ComponentDidMount", () => {

        it("Calls to fetch", async () => {
            const mockedToken = "VALUE";
            const expectedOptions = { 
                method: 'GET',
                headers:
                { Accept: 'application/json',
                    'Content-Type': 'application/json',
                    'x-access-token': mockedToken } 
                }
            fetchMock.get(BACKEND_IP + '/central/nurses', JSON.stringify([ mockedNurse, mockedNurse ]));

            const navigation = {
                getParam: jest.fn(key => mockedToken)
            }

            renderer.create(
                <GlobalScreen navigation={navigation}/>
            );

            expect( fetchMock.lastOptions() ).toEqual(expectedOptions); 
        });

        it("Catches a fetch error", async () => {
            const mockedToken = "VALUE";
            fetchMock.get(BACKEND_IP + '/central/nurses', 
                new Promise((res, rej) => rej("ERROR")),
                { overwriteRoutes: true });

            const navigation = {
                getParam: jest.fn( key => mockedToken ),
                navigate: jest.fn( name => {} )
            }

            const rendered = renderer.create(
                <GlobalScreen navigation={navigation}/>
            );

            await rendered.root._fiber.stateNode.componentDidMount()

            const renderedObj = rendered.root._fiber.stateNode;
            expect( renderedObj.state.error ).toBe(true);
        });
    });

    describe("View", () => {
        it("Shows 'Se ha producido un error...' when the fetch failed", async () => {
            
            fetchMock.get(BACKEND_IP + '/central/nurses', 
                new Promise((res, rej) => rej("ERROR")),
                { overwriteRoutes: true });

            const mockedToken = "VALUE";

            const navigation = {
                getParam: jest.fn( key => mockedToken ),
                navigate: jest.fn( name => {} )
            }

            const rendered = renderer.create(
                <GlobalScreen navigation={navigation}/>
            );

            await rendered.root._fiber.stateNode.componentDidMount()

            const loadingView = rendered.toJSON().children[0];
            const loadingText = loadingView.children[0];
            expect( loadingText ).toBe(" Se ha producido un error ");
            expect( rendered    ).toMatchSnapshot();
        });

        it("Shows Loading animation when the turns are not loaded", () => {
            const rendered = renderer.create(
                <GlobalScreen />
            ).toJSON();

            expect( rendered    ).toMatchSnapshot();
        });

        it("Shows everything when the turns are loaded", () => {
            const rendered = renderer.create(
                <GlobalScreen />
            )

            const renderedObj = rendered.root._fiber.stateNode;
            renderedObj.setState({
                usersLoaded: true,
                nurses: [ mockedNurse, mockedNurse ]
            });

            const titleView = rendered.toJSON().children[0];
            const titleText = titleView.children[0];
            expect( titleText         ).toBe("Turno de Febrero");
            expect( rendered.toJSON() ).toMatchSnapshot();
        });
    });

    describe("Children props", () => {
        it("TurnTable's setMonth method calls setState", async () => {
            // MOCKING CALLED FUNCTION SO THE NUMBER
            // OF CALLS IS CONTROLLED
            const mockedFn = jest.fn();

            // MOCKING PROPS SO THE COMPONENT IS 
            // CORRECTLY CREATED
            const mockedProps = {
                nurses: [ mockedNurse, mockedNurse ],
                navigation: {
                    getParam: () => "VALUE"
                }
            }
            const g = new GlobalScreen(mockedProps);
            
            // MOCKING THE STATE SO THE NORMAL VIEW
            // IS DISPLAYED INSTEAD OF 'Cargando...'
            g.state = {
                nurses: [ mockedNurse, mockedNurse ],
                usersLoaded: true
            }

            // MOCKING THE FUNCTION THAT IT IS EXPECTED
            // TO BE CALLED
            g.setState = mockedFn;

            // RENDERING
            const rendered        = g.render();

            // OBTAINING THE METHOD THAT SHOULD CALL
            // THE FUNCTION
            const viewContainer   = rendered.props.children[1];
            const testedComponent = viewContainer.props.children[1];
            const testedMethod    = testedComponent.props.setMonth;

            // CALLING THE METHOD AND ASSERTING THAT IT
            // CALLS TO THE CORRECT FUNCTION
            testedMethod();
            expect( mockedFn ).toBeCalled();
        });
    });
});