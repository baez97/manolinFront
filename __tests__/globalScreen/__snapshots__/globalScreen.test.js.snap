// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalScreen tests View Shows 'Se ha producido un error...' when the fetch failed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bee6ef",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 17.5,
    }
  }
>
  <Text>
     Se ha producido un error 
  </Text>
</View>
`;

exports[`GlobalScreen tests View Shows Loading animation when the turns are not loaded 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bee6ef",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 17.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      animating={true}
      animationDuration={1200}
      animationEasing={[Function]}
      color="#067bdb"
      interaction={true}
      renderComponent={[Function]}
      style={
        Object {
          "height": 262.5,
          "width": 262.5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#067bdb",
              "borderRadius": 131.25,
              "height": 262.5,
              "opacity": 0.5,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 262.5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#067bdb",
              "borderRadius": 131.25,
              "height": 262.5,
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 0.4,
                },
              ],
              "width": 262.5,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`GlobalScreen tests View Shows everything when the turns are loaded 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#bee6ef",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 17.5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#332554",
        "fontFamily": "montserrat-extra-bold",
        "fontSize": 38.95833442608516,
        "marginTop": 52.5,
        "textAlign": "left",
        "width": 712.5,
      }
    }
  >
    Turno de Febrero
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 17.5,
        "width": 712.5,
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 35,
          "marginTop": 78.75,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        Object {
          "borderRadius": 35.5,
          "elevation": 15,
          "overflow": "hidden",
          "width": 531.5,
        }
      }
    />
  </View>
</View>
`;
